src/cmsisPosix_Semaphore.c

```diff
@@ -0,0 +1,128 @@
+#define _GNU_SOURCE
+#include <semaphore.h>
+#include <pthread.h>
+#include <errno.h>
+
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include "cmsis_os2.h"
+#include "cmsisPosix_Config.h"
+
+// Forward declarations
+void cp_timeoutToTimespec(uint32_t timeout, struct timespec *ts);
+
+// Structure to hold POSIX semaphore and CMSIS attributes
+typedef struct
+{
+    const char *name;   // Name of semaphore
+    sem_t sem;          // POSIX semaphore
+} cp_semaphoreData_t;
+
+osSemaphoreId_t osSemaphoreNew(uint32_t max_count, uint32_t initial_count, const osSemaphoreAttr_t *attr)
+{
+    if (max_count == 0) {
+        return NULL;
+    }
+
+    cp_semaphoreData_t *semaphore = malloc(sizeof(cp_semaphoreData_t));
+    if (semaphore == NULL) {
+        return NULL;
+    }
+
+    if (sem_init(&semaphore->sem, 0, max_count) != 0) {
```

Comment:
the third argument of sem_init is the init value and not max value, should create other way to have max value (storing it in the data struct and tracking it) as POSIX not support this feature